lib/client/connection.js

```diff
@@ -753,7 +759,7 @@ Connection.prototype._handlePingPong = function(error) {
 Connection.prototype._initialize = function(message) {
   if (this.state !== 'connecting') return;
 
-  if (message.protocol !== 1) {
+  if (message.protocol !== protocol.major) {
```

Comment:
Confirming that we intend for a client to only support servers on the same major protocol version, disallowing a newer client to connect to a server on an older major version?

Reply:
Confirmed that is intended after discussion. That situation would only potentially happen during a rolling update that includes a major protocol update; or a server version rollback under the same situation.